.github/workflows/pr-functional-tests.yaml

```diff
@@ -206,5 +206,8 @@ jobs:
           TARGET_TAG: ${{ steps.check_version_in_PR.outputs.PR_version }}
           TARGET_COMMIT: ${{ steps.main_sha.outputs.origin_main_sha }}
         run: |
-          git tag "${TARGET_TAG}" "${TARGET_COMMIT}"
-          git push --force-with-lease origin "${TARGET_TAG}"
+          gh api \
+            --method POST \
+            "/repos/${GITHUB_REPOSITORY}/git/refs" \
+            -f "ref=refs/tags/${TARGET_TAG}" \
+            -f "sha=${TARGET_COMMIT}"
```

Comment:
Tested the command, it works as expected. I just want to note that the documentation mentions a different way to [create tags](https://docs.github.com/en/rest/git/tags?apiVersion=2022-11-28#create-a-tag-object) or to [create refs](https://docs.github.com/en/rest/git/refs?apiVersion=2022-11-28#create-a-reference).

Reply:
Indeed, the documentation mentions annotated tags which involves a ref creation - but as far as I could read (from the source code of the external action we used), we weren't creating an annotated tag there. If we end up needing to do so, I'll follow up with a patch to add that as well.

Reply:
TIL about the difference between lightweight and annotated tags